### DISCUSSION TITLE
More questions about precip values

### DISCUSSION THREAD

**Discussion by fcastellon** [weather-gov/api#683] [General]
Hello all:

My question has to do with the interpretation of the 'quantitativePrecipitation' and 'snowfallAmount' properties that you get from the "https://api.weather.gov/gridpoints/lat,lon" url. The information provided by @netbymatt in [this thread](https://github.com/weather-gov/api/discussions/647) is great and helped to clear some of my questions. However. I am currently querying for the amount of snow that will fall in a specific area and I find that under the quantitativePrecipitation property I see this:
                {
                    "validTime": "2023-12-19T00:00:00+00:00/PT6H",
                    "value": 0.254
                },
                {
                    "validTime": "2023-12-19T06:00:00+00:00/PT6H",
                    "value": 0.50800000000000001
                },

Then when I check the snowfallAmount for the same time I see this:
                {
                    "validTime": "2023-12-19T00:00:00+00:00/PT6H",
                    "value": 2.54
                },
                {
                    "validTime": "2023-12-19T06:00:00+00:00/PT6H",
                    "value": 5.0800000000000001
                },
Note that the values are off by a decimal place but the units for both measurements are expressed in mm per the corresponding 'uom'. 

My first question is which property is expressing the correct value? 

My second question is since 'quantitativePrecipitation' is already giving me the amount expected for that period and I can use the 'weather-->values-->value-->weather' property also provided in the JSON file to see if weather is 'snow or 'snow_showers' for that specified period (or rain, rain_showers, or ice_crystals) why is there a need for the 'snowfallAmount' or even the 'iceAccumulation' properties. What am I missing that these properties provide that I can't calculate from the use of weather and quantitativePrecipitation?

Thanks.

**Comment by dwhitemv25** [comment]
For your first question, they are both correct. `snowfallAmount` is the amount of _frozen_ precip expected, and `quantitativePrecipitation` expresses the amount of _liquid_ precipitation expected, including the amount of melted frozen precip. The ratio between the two is known as the Snow-Liquid Ratio (SLR), and in situations with decent snow production is usually around 10:1. In your case, 0.1" of snow is forecast, which if melted would produce 0.01" of liquid water. Low SLRs (like 5:1) represent "wet snow" which can drop lots of water in short periods. High SLRs give fluffy powdery snow which can build up into large drifts.

For your second question, its generally useful for weather forecasting to tell people how _much_ of the weather to expect. It's a big difference for city planners to know to expect 0.1" of snow (a dusting) vs 10" (gas up the snowplows) or 30" (better schedule some extra snowplow shifts)! 

Ice accumulation (from freezing rain or spray) can't be simply intuited from QPF and predominant weather, since it relies on knowing precise details about temperatures in the air column. And similarly to the snow example, large amounts of ice accumulation represent significant public safety concerns and it is worthwhile to call out the amount specifically so everyone can be prepared.

**Reply by fcastellon** [reply]
@dwhitemv25 thank-you for the detailed response. This helps me to make better sense of the numbers in the different categories. It's much appreciated.

Perhaps I didn't look in the right places but is there somewhere on this site that describes each of the fields provided by the API response (at least the ones that are not intuitive) in more detail like you have done here?

**Reply by dwhitemv25** [reply]
When you say "fields", do you mean the weather elements (the keys in `properties`)?

Some of the `gridpoints` elements are explained in the [gridpoints documentation](https://weather-gov.github.io/api/gridpoints).

Others can be found in the [NDFD element descriptions](https://digital.weather.gov/staticpages/definitions.php).

Other non-weather fields should be in the [API specification tab](https://www.weather.gov/documentation/services-web-api).

Feel free to ask about anything not covered above. Google can usually help here, too; the element definitions had to come from some previous research and the papers are usually available online somewhere.

**Reply by fcastellon** [reply]
@dwhitemv25 thanks again, this is a solid amount of information, particularly the https://digital.weather.gov/staticpages/definitions.php link.

thank-you!

**Reply by dwhitemv25** [reply]
For the record, the `gridpoints` data comes from WFO forecast packages and not NDFD, so some of the definitions from that page may not be 100% accurate. It should at least point you in the right direction.
